proposals/20240606-jwt-handling.md

```diff
@@ -0,0 +1,125 @@
+# Standardise JWTs Usage
+
+**Authors**: Kimonas Sotirchos @kimwnasptd
+
+## Requirements
+- OIDC logic, via [Istio external authoriser](https://istio.io/latest/docs/tasks/security/authorization/authz-custom/), should be adding `id_token` to http requests in `Authorization: Bearer <token>` headers
+## Scope
+This proposal aims to standardise how the Kubeflow backends should be handling the user information (name of user, groups they belong to), living in JWTs and http headers.
+
+This proposal takes as a requirement that users should be able to use K8s Tokens as `Authorization: Bearer <token>` headers in their request from inside the cluster.
```

Comment:
> At end, K8s ServiceAccount Tokens, Dex Tokens, even GitLab Tokens are JWTs and the difference is only who provides those and what claims are available.

Yes! That's why I like the proposal of off-loading the JWT handling in the service-mesh. Because then the apps can be completely.

Pushed a commit to try and clarify this and the multiple issuers. LMK what you think
7a4aeb91

Reply:
I like the updated version. Thanks.